v2/emailpassword/nextjs/setting-up-frontend.mdx

```diff
@@ -30,19 +34,19 @@ import AppInfoForm from "/src/components/appInfoForm"
 ```tsx title="pages/auth/[[...path]].tsx"
 import React, { useEffect } from 'react'
 import dynamic from 'next/dynamic'
-import SuperTokens from 'supertokens-auth-react'
+import { ^{recipePreBuiltUINameCapitalLetters} } from "supertokens-auth-react/recipe/^{codeImportRecipeName}/preBuiltUI";
 import { redirectToAuth } from 'supertokens-auth-react'
 
-const SuperTokensComponentNoSSR = dynamic<React.ComponentProps<typeof SuperTokens.getRoutingComponent>>(
-  new Promise((res) => res(SuperTokens.getRoutingComponent)),
+const SuperTokensComponentNoSSR = dynamic<React.ComponentProps<typeof ^{recipePreBuiltUINameCapitalLetters}.getRoutingComponent>>(
+  new Promise((res) => res(^{recipePreBuiltUINameCapitalLetters}.getRoutingComponent)),
   { ssr: false }
 )
 
 export default function Auth() {
 
   // if the user visits a page that is not handled by us (like /auth/random), then we redirect them back to the auth page.
   useEffect(() => {
-    if (SuperTokens.canHandleRoute() === false) {
+    if (^{recipePreBuiltUINameCapitalLetters}.canHandleRoute() === false) {
       redirectToAuth()
     }
```

Comment:
this will cause issues.. cause if someone is using email verification, and they haven't added the route checking for email verification here (cause the snippet here doesn't say that), then it will cause them to be redirected to the auth page.

Can we change the lib to have something like `supertokens.canHandleRoute()` instead of doing per recipe canHandleRoute?

Reply:
We could technically work our way around that. Still, even then, I think it'd be strange: even if you are never adding/rendering the pre-built passwordless UI, it will return true, suggesting we'd handle it when that's not necessarily true.